scripts/migrate_registration_logs.py

```diff
@@ -0,0 +1,45 @@
+import logging
+
+from modularodm import Q
+
+from website.models import NodeLog
+from website.app import init_app
+
+from scripts import utils as script_utils
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def get_targets():
+    logs = NodeLog.find(
+            Q('action', 'eq', 'embargo_approved') |
+            Q('action', 'eq', 'embargo_initiated') |
+            Q('action', 'eq', 'retraction_approved') |
+            Q('action', 'eq', 'retraction_initiated') |
+            Q('action', 'eq', 'registration_initiated')
+    )
+    return logs
+
+
+def migrate_log(logs):
+    for log in logs:
+        log.params['node'] = log.node.registered_from_id
+        log.save()
+        logger.info('Finished migrate log {}: registration action {} of node {}'.format(log._id, log.action, log.node))
+
+
+def main(dry_run):
+    logs = get_targets()
+    if not dry_run:
```

Comment:
It would be best to run `main` in a `TokuTransaction`, not only because this allows you to actually run `migrate_logs` during a dry run and roll back the transaction afterwards, but also because it won't leave the database in a half-migrated state if another error does occur:

```python
from framework.transactions.context import TokuTransaction
[...]
def main(dry_run):
    logs = get_targets()
    migrate_log(logs)
    logger.info('Finished migrate {} logs '.format(len(logs)))
    if dry_run:
        raise RuntimeError('Dry Run -- Transaction rolled back')
       
if __name__ == '__main__':
    [...setup...]
    with TokuTransaction():
        main(dry_run=dry_run)

```